src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -180,22 +228,25 @@ private async Task StartDashboardAsync(CancellationToken cancellationToken = def
         };
 
         await kubernetesService.CreateAsync(dashboardExecutable, cancellationToken).ConfigureAwait(false);
-        await WaitForHttpSuccessOrThrow(dashboardUrl, GetDashboardWaitTime(), cancellationToken).ConfigureAwait(false);
+        await WaitForHttpSuccessOrThrow(dashboardUrl, DashboardAvailabilityTimeoutDuration, cancellationToken).ConfigureAwait(false);
     }
 
-    private static TimeSpan GetDashboardWaitTime()
+    private static TimeSpan DashboardAvailabilityTimeoutDuration
     {
-        if (Environment.GetEnvironmentVariable("DOTNET_ASPIRE_DASHBOARD_TIMEOUT") is { } timeoutString && int.TryParse(timeoutString, out var timeoutInSeconds))
+        get
         {
-            return TimeSpan.FromSeconds(timeoutInSeconds);
-        }
-        else
-        {
-            return TimeSpan.FromSeconds(DefaultDashboardWaitTimeInSeconds);
+            if (Environment.GetEnvironmentVariable("DOTNET_ASPIRE_DASHBOARD_TIMEOUT") is { } timeoutString && int.TryParse(timeoutString, out var timeoutInSeconds))
```

Comment:
Perhaps the env var should have a unit in its name (`*_SECONDS`). I could easily assume it was millis.